app/src/main/java/ar/com/wolox/android/example/network/repository/NewsRepository.kt

```diff
@@ -0,0 +1,18 @@
+package ar.com.wolox.android.example.network.repository
+
+import ar.com.wolox.android.example.network.services.NewsService
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices
+import ar.com.wolox.wolmo.networking.retrofit.handler.NetworkRequestHandler
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+class NewsRepository @Inject constructor(private val retrofitServices: RetrofitServices) {
+
+    private val service: NewsService
+            get() = retrofitServices.getService(NewsService::class.java)
+
+    suspend fun getNews(accesstoken: String?, client : String?, uid : String?, page : Int) = withContext(Dispatchers.IO) {
+        NetworkRequestHandler.safeApiCall { service.getNews(accesstoken,client,uid, page) }
```

Comment:
Here I challenge you: How could you do to send the `access_token`, `client` and `uid` in every request without passing them as parameters of the function? (Hint: With Retrofit!). Ask me anything you need with this question!